Makefile

```diff
@@ -9,6 +9,11 @@
 #     For correctness and performance tests this include halide build time and run time. For
 #     the tests in test/generator/ this times only the halide build time.
 
+# Disable built-in makefile rules for all apps to avoid pointless file-system
+# scanning and general weirdness resulting from implicit rules.
+MAKEFLAGS += --no-builtin-rules --no-builtin-variables
```

Comment:
This makes make intentionally dumber. It prevents a bunch of silly built-in rules from running and failing. The two main effects are a slight increase in performance, and also when you typo a target name, or try for a target that doesn't exist, it errors out instead of doing something weird based on implicit magic. 

For example in iir_blur, if I say "make bin/host/iir_blur.generator" it makes that file. Correct. But if I say "make iir_blur_generator" it finds iir_blur_generator.cpp, then decides to invoke a c++ compiler on it with the wrong set of flags to try to compile that file as a standalone tool into the current directory. You get a spew of linker errors. With implicit rules deleted, it instead says "no rule to make target iir_blur_generator", which is more helpful.